src/common/selectors/deployComponentsSelector.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+import { QueryResult } from 'jsforce';
+import { DeployComponent } from '../types';
+
+export async function selectDeployComponentsByAsyncOpId(
+  con: Connection,
+  asyncOpId: string
+): Promise<DeployComponent[]> {
+  const queryStr = `SELECT sf_devops__Source_Component__c, sf_devops__Operation__c, sf_devops__File_Path__c
+                    FROM sf_devops__Deploy_Component__c  
+                    WHERE sf_devops__Deployment_Result__r.sf_devops__Status__c = '${asyncOpId}'`;
+  const resp: QueryResult<DeployComponent> = await con.query(queryStr);
+
+  resp.records.forEach((record) => {
```

Comment:
I think we should move this logic from the selector to the caller in case this method is called from other places in the future.

Reply:
It can be used from other places, I thought this was the best option because in most cases we want the type and name separately. If we move this logic outside potentially we are going to have that logic called from different places.